relay-single.ts

```diff
@@ -138,9 +138,22 @@ export class SingleRelayConnection implements Subscriber, SubscriptionCloser, Ev
                     messsage.type == "OtherError" || messsage.type == "closed"
                 ) {
                     if (messsage.type != "closed") {
-                        this.error = messsage.error;
+                        if (messsage.error instanceof Error) {
+                            this.error = messsage.error;
+                        } else if (typeof messsage.error == "string") {
+                            this.error = new Error(messsage.error);
+                        } else {
+                            console.error(messsage);
+                            this.error = new Error(messsage.error.error);
+                        }
+                    }
+                    console.log("connection error", messsage);
+                    if (messsage.type == "closed") {
+                        // https://www.rfc-editor.org/rfc/rfc6455.html#section-7.4
+                        if (messsage.event.code == 1011) {
+                            return new AuthError(messsage.event.reason);
+                        }
```

Comment:
According to the table found at https://www.iana.org/assignments/websocket/websocket.xml#close-code-number, it may be more appropriate to use 3000(Unauthorized).

Reply:
ok